### PR TITLE
[4] Start and Finished featured article settings cleared when saving article at frontend

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #38961 .

### Summary of Changes
 Get featured up & down info

### Testing Instructions
edit a featured article from frontend

### Actual result BEFORE applying this Pull Request
featured up  & down data cleaned

### Expected result AFTER applying this Pull Request
no more data loss

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by WoodyF4u** [comment]
I was the person who started #38961
I have seen that it is closed.
We are asked to test it in this PR.
But is there something to test anywhere? Where can I find it?

**Comment by alikon** [comment]
as usual
for more info https://docs.joomla.org/Testing_Joomla!_patches/en

**Comment by WoodyF4u** [comment]
I have tested this item :white_check_mark: successfully on b77c583ad5f6a7ee58a8d90f302a39d603759de6<br /><br />I have tested the patch on a copy of a live site.
The result of the test was good.
Without a patch there was no field information for the beginning and end of the featured articles.
It works fine with the patch.
Even after saving in the frontend, the data is the same on the frontend and backend.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

File: components/com_content/src/Model/FormModel.php [hunk 1]
```diff
@@ -152,6 +152,19 @@ public function getItem($itemId = null)
...
+        // Get featured up & down info
+        $db    = $this->getDatabase();
+        $query = $db->getQuery(true)
+            ->select($db->quoteName(['featured_up', 'featured_down']))
+            ->from($db->quoteName('#__content_frontpage'))
+            ->where($db->quoteName('content_id') . ' = :articleId')
+            ->bind(':articleId', $value->id, ParameterType::INTEGER);
+        $db->setQuery($query);
+
+        $result = $db->loadObject();
```
Comment (HLeithner) [inline components/com_content/src/Model/FormModel.php]: this could be null

**Comment by WoodyF4u** [comment]
Is it possible to see this solutuion in the next Joomla 4 release?
For example in 4.2.5?

File: components/com_content/src/Model/FormModel.php (hunk 1 above)
Comment (alikon) [reply components/com_content/src/Model/FormModel.php]: checked

**Comment by joomdonation** [comment]
I think you can improve the current code a a little by doing the same with the code in `Article` model https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_content/src/Model/ArticleModel.php#L402-L430

- Move the code inside `if ($itemId)` block
- Only execute the query to get data for featured article only if `($value->featured)`, same with https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_content/src/Model/ArticleModel.php#L409

**Comment by Scrabble96** [comment]
What happened to this PR? Why did it not appear in 4.2.6? More testing on the front and backend shows that:

1. On front end - i) add start and finish dates for Featured, ii) 'Save' (the dates apparently disappear) then 'Cancel', or add the dates and then 'Save and Close', iii) check in backend: Featured dates are shown;
However...
2. On front end - i) add start and finish dates for Featuring, ii) 'Save' (the dates apparently disappear), iii) continue editing content, iv) Click 'Save and Close', v) check in backend: Featured dates are lost because they have been cleared earlier in the 'Save' operation.

**Comment by alikon** [comment]
as usual it needs 2 successfull tests and then it's up mantainers / Release leaders to decide

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on d04129d28586b1608c82a1997a0616b39471899e<br /><br />The PR solves the issue. Further improvement could be made to avoid repeating same code for both frontend and backend, but it could be done in separate PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by joomdonation** [comment]
@Scrabble96 or @WoodyF4u If you have time, please test this PR so that we can get the issue fixed for next release.

**Comment by Scrabble96** [comment]
I have tested the patch successfully using the Joomla! Patch Tester component on a development site online. I can't see a way to submit my test results, though.

**Comment by joomdonation** [comment]
You can go to https://issues.joomla.org/tracker/joomla-cms/38975 , click on **Test this** button at the top to report test result. See the attached screenshot for more details
test

**Comment by Scrabble96** [comment]
I have tested this item :white_check_mark: successfully on d04129d28586b1608c82a1997a0616b39471899e<br /><br />Tested as per my comment 2022/12/29<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by joomdonation** [comment]
RTC. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by WoodyF4u** [comment]
I just tested it successfully as well.
I saw that two successful tests have already been done.
So I can't officially confirm it anymore, but I agree with this patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by WoodyF4u** [comment]
I just tested it successfully as well.
I saw that two successful tests have already been done.
So I can't officially confirm it anymore, but I agree with this patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by brianteeman** [comment]
@WoodyF4u you can still record your successful test<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by WoodyF4u** [comment]
I have tested this item :white_check_mark: successfully on d04129d28586b1608c82a1997a0616b39471899e<br /><br />My test was succesfull.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by WoodyF4u** [comment]
Hi Brian,
Done.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38975">issues.joomla.org/tracker/joomla-cms/38975</a>.</sub>

**Comment by fancyFranci** [comment]
Thanks a lot!

**Pull request merged**